pkg/certificate/reconciler.go

```diff
@@ -0,0 +1,48 @@
+package certificate
+
+import (
+	"context"
+	"time"
+)
+
+// CheckAndUpdate checks for a MRC to reach a desired condition and when the condition is met,
+// it updates the status and terminates the loop
+func (r *MRCReconciler) CheckAndUpdate(ctx context.Context, mrcClient MRCClient, checkInterval time.Duration) {
```

Comment:
so it seems this, and CheckStatus are the only 2 new methods.

The certificate package is getting a little complicated with the number of types we have, and the layers of composition required. In general small building blocks are favorable, but we're starting to create a web of building blocks that on first glance is hard to grok.

I wonder if it makes more sense to put these methods on one of the other existing structs for now?

Reply:
When we get rid of the MRCCompatClient I think the certificate package will become less complex, but I agree that the MRCReconciler adds another layer of complexity.

There are several different MRC reconciliation loops. Each loop is checking for different status conditions and will make different updates to the MRC.  I had envisioned having unique implementations of `updateStatus` and `checkStatus` for each loop. The goal of the `MRCReconciler` was to define a reconciliation pattern applicable to all loops while also providing the ability to uniquely specify how to check and update the MRC. The `MRCReconciler` would be configured in response to an MRC event.

I am not sure there is a struct in `pkg/certificate` where it makes sense to put these methods. I'd say the `MRCComposer` would make sense but that would result in circular dependency if we want to call `checkAndUpdate` in `manager.go`

Reply:
@steeling Let me know if this makes sense or if I should find a way to avoid a new MRCReconciler struct :)

Reply:
That sounds pretty complex. To dynamically change the method that is called based on the event type, when that method is a field on the struct, instead of being explicit about `UpdateStatusForNewReadyState`, or something similar.

That hides some "magic", and forces a new reader to dig around a bit more to discover what is happening.

What if we simply put the reconciler logic on the MRCClient?

Reply:
It’s really difficult to sign off on this PR as is. The MRCReconciler is never instantiated, nor is there an interface to demonstrate how it is used. 

Instead, can we hook it into whatever calls CheckAndRotate?

A PR should come in before this that demonstrates how this is plumbed in, even if the reconciler loop does nothing. Then we can flesh out the implementation. 

Or make this a full implementation that has update logic. But right now we’ve created a middle piece that doesn’t connect on either end so it’s hard to see how the roads will meet